src/utils/signal.ts

```diff
@@ -15,6 +17,11 @@ export interface CanDisconnect {
 export class SignalHandle {
     private constructor(target: CanDisconnect, id: number) {
         this.target = target;
+        if (target instanceof Clutter.Actor) {
+            target.connect('destroy', () => {
+                this.disconnect();
```

Comment:
If this runs, and later SignalObserver.clear runs, it will log a bunch of `Signal is already disconnected`. Might not be ideal? But I guess it's not a huge deal.

Reply:
we can live with it